File: src/Exercises/ContainsDuplicate.cs
namespace Exercises
{
    public class ContainsDuplicate
    {

        public bool containsDuplicate(int[] numbers)
        {
            HashSet<int> uniqueNumbers = new HashSet<int>();
            foreach (int number in numbers)
            {
                if (uniqueNumbers.Contains(number)) return true;
                uniqueNumbers.Add(number);
            }
            return false;
        }

        // Este código utiliza um HashSet para armazenar elementos únicos.
        // Ele percorre os elementos na coleção "numbers" um por um na ordem em que aparecem.
        // Para cada elemento, verifica se já existe no HashSet.
        // Se o elemento já estiver presente, isso significa que há um número duplicado e retorna "true".
        // Caso contrário, adiciona o elemento ao HashSet.
        // Se não houver números duplicados em toda a coleção, o loop termina e o método retorna "false".
    }
}

File: go/exercises/code/ContainsDuplicate.go
package code

// ContainsDuplicate verifica se há números duplicados no slice fornecido
func ContainsDuplicate(numbers []int) bool {
	// Cria um mapa para armazenar números únicos
	uniqueNumbers := make(map[int]bool)

	// Percorre cada número no slice "numbers"
	for _, number := range numbers {
		// Verifica se o número já está presente no mapa
		if _, exists := uniqueNumbers[number]; exists {
			// Se o número já existir, encontramos um número duplicado
			return true
		}

		// Caso contrário, adiciona o número ao mapa
		uniqueNumbers[number] = true
	}

	// Se nenhum número duplicado foi encontrado, retorna "false"
	return false
}
